my-webxr-app/src/repository/DbRepository.tsx

```diff
@@ -128,9 +241,37 @@ export default class DbRepository extends Dexie implements Repository {
     return dataPoints;
   }
 
-  /*
-        closeConnection closes the connection to the database
-         */
+  /**
+   * Retrieves all column names from the raw data table in the database.
+   * This will include all columns that have been added from the CSV.
+   *
+   * @returns {Promise<string[]>} A promise resolves to an array of column names.
+   */
+  async getAllColumnNames(): Promise<string[]> {
+    const rawColumnNames: string[] = [];
+    const columns = await this.rawColumns.toArray();
+    const columnNames = columns.map((column) => column.name);
+    rawColumnNames.push(...columnNames);
+    return Promise.resolve(rawColumnNames);
+  }
+
+  /**
+   * Retrieves all numeric column names from the stat data table in the database.
+   * This assumes that the stats data table (look-up table) contains only numeric columns.
+   *
+   * @returns {Promise<string[]>} A promise resolves to an array of column names.
+   */
+  async getNumericColumnNames(): Promise<string[]> {
+    const numericColumnNames: string[] = [];
+    const columns = await this.statsColumns.toArray();
+    const columnNames = columns.map((column) => column.name);
+    numericColumnNames.push(...columnNames);
+    return Promise.resolve(numericColumnNames);
```

Comment:
Do both functions, getNumericColumnNames() and getAllColumnNames(), resolve to Promise? Are both called simultaneously or is one overwritten?

Reply:
Not sure if i got the question. 

Get allColumnNames return all the column names in the raw table -- all the header in the original headers in csv.

Get numericColumnNames return all the column names in the stat table, which are all numbers, i plan to use this for user to select columns to perform pca on.

Reply:
They both resolve to promise which are both array of strings

Reply:
Oh ok, thanks for clarifying!